manager/manager.go

```diff
@@ -0,0 +1,1210 @@
+	}
+
+	// Match maps
+	for _, managerMap := range m.Maps {
+		spec, ok := m.collectionSpec.Maps[managerMap.Name]
+		if !ok {
+			errors.Wrapf(ErrUnknownSection, "couldn't find map at maps/%s", managerMap.Name)
+		}
+		spec.Contents = managerMap.Contents
+		spec.Freeze = managerMap.Freeze
+		managerMap.arraySpec = spec
+	}
+
+	// Match perfmaps
+	for _, perfMap := range m.PerfMaps {
+		spec, ok := m.collectionSpec.Maps[perfMap.Name]
+		if !ok {
+			errors.Wrapf(ErrUnknownSection, "couldn't find map at maps/%s", perfMap.Name)
+		}
+		perfMap.arraySpec = spec
+	}
+	return nil
+}
+
+func (m *Manager) activateProbes() {
+	for _, mProbe := range m.Probes {
+		shouldActivate := false
+		for _, probe := range m.options.ActivatedProbes {
+			if probe == mProbe.Section {
+				shouldActivate = true
+			}
+		}
+		if shouldActivate || len(m.options.ActivatedProbes) == 0 {
+			mProbe.Enabled = true
+		}
+	}
+}
+
+// editConstants - Edit the programs in the CollectionSpec with the provided constant editors. Tries with the BTF global
+// variable first, and fall back to the asm method if BTF is not available.
+func (m *Manager) editConstants() error {
+	// Start with the BTF based solution
+	rodata := m.collectionSpec.Maps[".rodata"]
+	if rodata != nil && rodata.BTF != nil {
+		consts := map[string]interface{}{}
+		for _, editor := range m.options.ConstantEditors {
+			consts[editor.Name] = editor.Value
+		}
+		return m.collectionSpec.RewriteConstants(consts)
+	}
+
+	// Fall back to the old school constant edition
+	for _, constantEditor := range m.options.ConstantEditors {
+
+		// Edit the constant of the provided programs
+		for _, id := range constantEditor.ProbeIdentificationPairs {
+			programs, found, err := m.GetProgramSpec(id)
+			if err != nil {
+				return err
+			}
+			if !found || len(programs) == 0 {
+				return errors.Wrapf(ErrUnknownSection, "couldn't find programSpec %v", id)
+			}
+			prog := programs[0]
+
+			// Edit program
+			if err := m.editConstant(prog, constantEditor); err != nil {
+				return errors.Wrapf(err, "couldn't edit %s in %v", constantEditor.Name, id)
+			}
+		}
+
+		// Apply to all programs if no section was provided
+		if len(constantEditor.ProbeIdentificationPairs) == 0 {
+			for section, prog := range m.collectionSpec.Programs {
+				if err := m.editConstant(prog, constantEditor); err != nil {
+					return errors.Wrapf(err, "couldn't edit %s in %s", constantEditor.Name, section)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// editMapSpecs - Update the MapSpec with the provided MapSpec editors.
+func (m *Manager) editMapSpecs() error {
+	for name, mapEditor := range m.options.MapSpecEditors {
+		// select the map spec
+		spec, exists, err := m.GetMapSpec(name)
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return errors.Wrapf(ErrUnknownSection, "failed to edit maps/%s: couldn't find map", name)
+		}
+		spec.Type = mapEditor.Type
+		spec.MaxEntries = mapEditor.MaxEntries
+		spec.Flags = mapEditor.Flags
```

Comment:
One gotcha here is that you must specify both `MaxEntries` and `Flags`. Otherwise the value from the ELF file will be overwritten with `0`. Is that intentional?

Reply:
I added a Flag so that you can specify which entries to update.